core/src/main/java/jenkins/model/item_category/ItemCategory.java

```diff
@@ -0,0 +1,112 @@
+package jenkins.model.item_category;
+
+import hudson.Extension;
+import hudson.ExtensionPoint;
+import hudson.model.ModelObject;
+import jenkins.model.Messages;
+
+/**
+ * A category for {@link hudson.model.Item}s.
+ *
+ * @since TODO
+ */
+public abstract class ItemCategory implements ModelObject, ExtensionPoint {
+
+    public static int MIN_WEIGHT = 0;
+
+    public static int MIN_TOSHOW = 1;
+
+    /**
+     * Identifier, e.g. "category-id-standaloneprojects", etc.
+     *
+     * @return the identifier
+     */
+    public abstract String getId();
+
+    /**
+     * The description in plain text
+     *
+     * @return the description
+     */
+    public abstract String getDescription();
+
+    /**
+     * Helpful to set the order.
+     *
+     * @return the weight
+     */
+    public abstract int getWeight();
+
+    /**
+     * Minimum number required to show the category.
+     *
+     * @return the minimum items required
+     */
+    public abstract int getMinToShow();
+
+    /**
+     * The default {@link ItemCategory}, if an item doesn't belong anywhere else, this is where it goes by default.
+     */
+    @Extension
+    public static final class UncategorizedCategory extends ItemCategory {
+
+        @Override
+        public String getId() {
+            return "itemcategory-uncategorized";
```

Comment:
@jglick I've removed the prefix because I don't want to spend time discussing this kind of things. I'll have to upgrade these two PRs: https://github.com/jenkinsci/maven-plugin/pull/62 and https://github.com/jenkinsci/matrix-project-plugin/pull/33 /cc @daniel-beck